src/App.vue

```diff
@@ -211,15 +209,11 @@ export default {
 			}
 			poll()
 		})
-		this.scrollParent = findScrollParent(this.$el.parentElement || this.$el.getRootNode().host)
-		if (this.scrollParent) {
-			this.scrollParentResizeObserver = new ResizeObserver(this.onScrollParentResize)
-			this.scrollParentResizeObserver.observe(this.scrollParent)
-			this.scrollParentWidth = this.scrollParent.offsetWidth
-		} else { // scrolling document
-			window.addEventListener('resize', this.onWindowResize)
-			this.onWindowResize()
-		}
+		const parent = this.$el.parentElement ?? this.$el.getRootNode().host
+		this.scrollParent = findScrollParent(parent) ?? parent ?? document.body
+		this.scrollParentResizeObserver = new ResizeObserver(this.onScrollParentResize)
+		this.scrollParentResizeObserver.observe(this.scrollParent)
+		this.scrollParentWidth = this.scrollParent.offsetWidth
```

Comment:
When updating the code to compute the width based on the parent, whenever possible, instead of the `document.body`, I also unified the mechanism to always use the `ResizeObserver`.